src/hooks/useFileSystemWatcher.tsx

```diff
@@ -0,0 +1,71 @@
+import { isDesktop } from 'lib/isDesktop'
+import { useEffect, useState, useRef } from 'react'
+
+type Path = string
+
+// Not having access to NodeJS functions has influenced the design a little.
+// There is some indirection going on because we can only pass data between
+// the NodeJS<->Browser boundary. The actual functions need to run on the
+// NodeJS side. Because EventEmitters come bundled with their listener
+// methods it complicates things because we can't just do
+// watcher.addListener(() => { ... }).
+
+export const useFileSystemWatcher = (
+  callback: (path: Path) => void,
+  dependencyArray: Path[]
+): void => {
+  // Track a ref to the callback. This is how we get the callback updated
+  // across the NodeJS<->Browser boundary.
+  const callbackRef = useRef<{ fn: (path: Path) => void }>({
+    fn: (_path) => {},
+  })
```

Comment:
I don't think this makes any material impact, but would [useCallback](https://react.dev/reference/react/useCallback) work the same here? like `const callbackRef = useCallback(callback, [])`.

Reply:
No I don't think so. useRef is used mostly because it gives us a permanent actual reference to an object in a React context. useCallback is to prevent re-computation of a function for various reasons. I _really_ needed a reference for this one.